### PR TITLE
#177 location_admin privileges + itcases and puzzle

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #177 

Added __tables_access check and integration tests for it.
Left puzzle for continuing implementation also on the reservations part.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#377](https://github.com/timelesslounge/timelessis/issues/377) is now in scope, role is `REV`

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#377` already assigned to @salgadobreno, can't assign to @vryazanov

**Comment by 0crat** [comment]
This pull request [#377](https://github.com/timelesslounge/timelessis/issues/377) is assigned to @salgadobreno[/z](https://www.0crat.com/u/salgadobreno), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/6b01437c-a2f8-47b7-8aae-079ae163e3a4); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_location_admin_privileges_test.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    )
+    table = Table(
+        id=1,
+        name="some table",
+        floor_id=floor.id,
+        x=40,
+        y=50,
+        width=320,
+        height=150,
+        status="available",
```
Comment (salgadobreno) [inline tests/integration/it_location_admin_privileges_test.py]: Status is defined as an `int` in `timeless/restaurants/models.py`, this is causing the Travis CI and tests to break.

File: tests/integration/it_location_admin_privileges_test.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
```
Comment (salgadobreno) [inline tests/integration/it_location_admin_privileges_test.py]: Same as previous comment.

**Comment by amihaiemil** [comment]
@salgadobreno Are you good in python and this Sqlalchemy ORM? I have a funny error now, see [log](https://travis-ci.org/timelesslounge/timelessis/builds/498388998#L1408). Basically, I'm not sure how to read the Floor, which is a linked relation to Table :)

**Comment by amihaiemil** [comment]
@salgadobreno it's fixed now, can you have a look? :)

**Comment by salgadobreno** [comment]
@amihaiemil looks good! I imagine you don't have access to Codeclimate details either? If so we need @emilianodellacasa to add us both to it.

**Comment by amihaiemil** [comment]
@salgadobreno 

> I imagine you don't have access to Codeclimate details either

I think I don't have access. But even if I do, I tried authenticating with Github and it requires access to my private repos as well, which I don't really like :/

**Comment by emilianodellacasa** [comment]
@amihaiemil now you have access to codeclimate

**Comment by amihaiemil** [comment]
@salgadobreno @emilianodellacasa fixed :)

**Comment by amihaiemil** [comment]
@emilianodellacasa wait! I haven't removed the puzzle :D

**Comment by amihaiemil** [comment]
@emilianodellacasa now it's really done :D

**Comment by emilianodellacasa** [comment]
@salgadobreno let me know if we can merge

**Comment by salgadobreno** [comment]
@rultor good to merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @salgadobreno[/z](https://www.0crat.com/u/salgadobreno), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#377](https://github.com/timelesslounge/timelessis/issues/377) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@salgadobreno According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@salgadobreno According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your code review comments by indicating an addressee in the beginning.

**Comment by salgadobreno** [comment]
@ypshenychka 

> @salgadobreno According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
> 
> > The code reviewer found at least three problems in the code.
> > Comments were mostly about design problems, not cosmetic issues.
> 
> Only two issues were found during code review.
> Please confirm that you'll try to find at least three major problems while future reviews.

I confirm

> @salgadobreno According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
> 
> > Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.
> 
> Please correct your code review comments by indicating an addressee in the beginning.

Done it

**Comment by ypshenychka** [comment]
@salgadobreno thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @salgadobreno[/z](https://www.0crat.com/u/salgadobreno)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**